### PR TITLE
[WIP] Test spectral

### PR DESCRIPTION
**Pull request by davidmeunier79**

### CODE REVIEW

**Comment by jasmainak** [comment]
@davidmeunier79 you need to remove all the merge commits to have a clean commit history. You can do a rebase for that. Something like:

```sh
$ git checkout test_spectral
$ git rebase -i HEAD~12
```

Never do a merge or pull. It's not good practice

**Comment by jasmainak** [comment]
@davidmeunier79 you have flake8 issues. You should use a flake8 plugin on your editor.

File: ephypype/tests/test_spectral.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+ts_mat_trials = np.random.rand(nb_trials,nb_ROI, time_length) * \
+    np.random.choice([-1, 1], size=(nb_trials, nb_ROI, time_length))
+              
+##################################### testing compute_spectral_connectivity ###################################
+def test_compute_spectral_connectivity_one_trial():
+    
+    """Test compute_spectral_connectivity plv with one trial."""
+    try:
+        res_plv = compute_spectral_connectivity(data = ts_mat, con_method = "plv", mode = "multitaper", fmin = fmin, fmax = fmax, sfreq = sfreq, gathering_method='mean')
+    except SystemExit:
```
Comment (jasmainak) [inline ephypype/tests/test_spectral.py]: There should be no `try-except` code in your tests ... you should use assert statements and pytest

Comment (davidmeunier79) [reply ephypype/tests/test_spectral.py]: how do you assert a function that return sys.exit exception? I tried to find but couldn't

Comment (jasmainak) [reply ephypype/tests/test_spectral.py]: you use `pytest.raises`

**Comment by annapasca** [comment]
Hi @davidmeunier79 !
I'm wondering before to add the example for connectivity pipeline I wait for this merge.
Is it ok?

File: ephypype/tests/test_spectral.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+############################### testing compute_and_save_multi_spectral_connectivity ##############################
+
+os.makedirs("/tmp/ephypype_test")
+    
+def test_compute_and_save_multi_spectral_connectivity():
+
+    compute_and_save_multi_spectral_connectivity(all_data = ts_mat_trials, con_method = "coh", mode = "multitaper", fmin = fmin, fmax = fmax, sfreq = sfreq, gathering_method='mean', save_dir = "/tmp/ephypype_test")
+
+if __name__ == '__main__':
```
Comment (jasmainak) [inline ephypype/tests/test_spectral.py]: there should be no "main" in the testing files

File: ephypype/tests/test_spectral.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    
+    print(res_max)
+    
+    assert np.all(res_mean <= res_max), "error, all mean values should be lower than max values"
+
+############################### testing compute_and_save_multi_spectral_connectivity ##############################
+
+os.makedirs("/tmp/ephypype_test")
+    
+def test_compute_and_save_multi_spectral_connectivity():
```
Comment (jasmainak) [inline ephypype/tests/test_spectral.py]: docstring missing

File: ephypype/tests/test_spectral.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        assert False
+      
+    #assert np.all(res_mean <= res_max), "error, all mean values should be lower than max values"
+        
+
+def test_compute_spectral_connectivity_mean_max():
+    
+    """Test compute_spectral_connectivity mean and max."""
+    
+    res_mean = compute_spectral_connectivity(data = ts_mat_trials, con_method = "coh", mode = "multitaper", fmin = fmin, fmax = fmax, sfreq = sfreq, gathering_method='mean')
```
Comment (jasmainak) [inline ephypype/tests/test_spectral.py]: pep8 violation

**Comment by davidmeunier79** [comment]
hopefully https://github.com/neuropycon/ephypype/pull/52 will do the job

**Pull request closed without merging**